runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkExecutableStageContext.java

```diff
@@ -35,8 +36,12 @@
     FlinkExecutableStageContext get(JobInfo jobInfo);
   }
 
-  static Factory factory() {
-    return FlinkDefaultExecutableStageContext.ReferenceCountingFactory.REFERENCE_COUNTING;
+  static Factory factory(FlinkPipelineOptions options) {
+    if ("[stage]".equals(options.getSdkWorkerParallelism())) {
```

Comment:
Perhaps define this constant in the Options?

Reply:
done